src/join.jl

```diff
@@ -424,10 +424,13 @@ If `dimmap` is not specified, it is determined automatically using index column
 names and types.
 """
 function broadcast(f::Function, A::IndexedTable, B::IndexedTable; dimmap=nothing)
+    out_T = Base.promote_op(f, eltype(A), eltype(B))
+    out = IndexedTable(similar(typeof(A.index), 0), similar(arrayof(out_T), 0))
```

Comment:
Seems to work if we replace `similar(typeof(A.index), 0)` with `similar(A.index, 0)`, though shouldn't this go inside the if statement below (because in the second case, you would want `B` instead of `A`?